src/v/storage/disk_log_impl.cc

```diff
@@ -953,6 +953,34 @@ offset_stats disk_log_impl::offsets() const {
     };
 }
 
+model::offset disk_log_impl::last_term_start_offset() const {
+    if (_segs.empty()) {
+        return {};
+    }
+
+    segment_set::type end;
+    segment_set::type term_start;
+    for (int i = (int)_segs.size() - 1; i >= 0; --i) {
```

Comment:
nit: this maybe cleaner with rbegin() and rend()?

Reply:
Ah, I see it's pulled from another method. Feel free to leave it as is, just a nit